scanpy/tools/_marker_gene_overlap.py

```diff
@@ -0,0 +1,181 @@
+    markers1: dict,
+    markers2: dict,
+):
+    """Calculate overlap count between the values of two dictionaries
+
+    Note: dict values must be sets
+    """
+    overlaps=np.zeros((len(markers1), len(markers2)))
+
+    j=0
+    for marker_group in markers1:
+        tmp = [len(markers2[i].intersection(markers1[marker_group])) for i in markers2.keys()]
+        overlaps[j,:] = tmp
+        j += 1
+
+    return overlaps
+
+
+def _calc_overlap_coef(
+    markers1: dict,
+    markers2: dict,
+):
+    """Calculate overlap coefficient between the values of two dictionaries
+
+    Note: dict values must be sets
+    """
+    overlap_coef=np.zeros((len(markers1), len(markers2)))
+
+    j=0
+    for marker_group in markers1:
+        tmp = [len(markers2[i].intersection(markers1[marker_group]))/
+               min(markers2[i], markers1[marker_group]) for i in markers2.keys()]
+        overlap_coef[j,:] = tmp
+        j += 1
+
+    return overlap_coef
+
+
+def _calc_jaccard(
+    markers1: dict,
+    markers2: dict,
+):
+    """Calculate jaccard index between the values of two dictionaries
+
+    Note: dict values must be sets
+    """
+    jacc_results=np.zeros((len(markers1), len(markers2)))
+
+    j=0
+    for marker_group in markers1:
+        tmp = [len(markers2[i].intersection(markers1[marker_group]))/
+               len(markers2[i].union(markers1[marker_group])) for i in markers2.keys()]
+        jacc_results[j,:] = tmp
+        j += 1
+
+    return jacc_results
+
+
+def marker_gene_overlap(
+    adata: AnnData,
+    key: str,
+    reference_markers: dict,
+    *,
+    method: Optional[str] = 'overlap',
+    normalize: Union[str, None] = None,
+    key_added: Optional[str] = 'marker_gene_overlap'
+):
+    """Calculate an overlap score between data-deriven marker genes and provided markers
+
+    Marker gene overlap scores can be quoted as overlap counts, overlap coefficients, or
+    jaccard indices. The method returns a pandas dataframe which can be used to annotate
+    clusters based on marker gene overlaps.
+
+    This function was written by Malte Luecken.
+
+    Parameters
+    ----------
+    adata : :class:`~anndata.AnnData`
+        The annotated data matrix.
+    key : `str`, optional (default: `rank_genes_groups`)
+        The key in `adata.uns` where the rank_genes_groups output is stored. This field
+        should contain a dictionary with a `numpy.recarray()` under the key 'names'.
+    reference_markers : `dict`, optional (default: `None`)
+
+    key_added : `str`, optional (default: `None`)
+
+    method : `{'overlap_count', 'overlap_coef', 'jaccard'}`, optional (default: `overlap_count`)
+
+    normalize : `{'reference', 'data', 'None'}`, optional (default: `None`)
+
+    Returns
+    -------
+    Updates `adata.uns` with an additional field specified by the `key_added`
+    parameter (default = 'marker_gene_overlap'). 
+
+    Examples
+    --------
+    >>> adata = sc.datasets.pbmc68k_reduced()
+    >>> 
```

Comment:
don't forget to finish this

Reply:
of course... it was just a copy-paste from embedding density for now ;).